File: sudhdkso/42583.cpp
#include <string>
#include <vector>
#include<queue>
#include<iostream>
using namespace std;

int solution(int bridge_length, int weight, vector<int> truck_weights) {
	int answer = 0, count = 0;
	int sum = 0;
	queue<int> q;
	while (1) {

		//cout << count << ":" << truck_weights.size() << " ";

		if (count > truck_weights.size() - 1)
			break;
		if (sum + truck_weights[count] <= weight) {
			if (q.size() < bridge_length) {
				sum += truck_weights[count];
				q.push(truck_weights[count]); count++;
				answer++;
			}
			else {
				sum -= q.front();
				q.pop();
			}

		}
		else {
			if (q.size() < bridge_length) {
				q.push(0);
				answer++;
			}
			else {
				sum -= q.front();
				q.pop();
			}
		}

	}

	return answer + bridge_length;
}

File: qwertycvb/42583.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    bridge_queue = [0 for _ in range(bridge_length)]

    while bridge_queue:
        answer += 1
        bridge_queue.pop(0)
        if truck_weights:
            if sum(bridge_queue) + truck_weights[0] <= weight:
                bridge_queue.append(truck_weights.pop(0))
            else:
                bridge_queue.append(0)
    return answer